test/src/unit-capi-array_schema.cc

```diff
@@ -1615,6 +1616,78 @@ TEST_CASE_METHOD(
   remove_temp_dir(local_fs.file_prefix() + local_fs.temp_dir());
 }
 
+TEST_CASE_METHOD(
+    ArraySchemaFx,
+    "C API: Test array schema setter/getter for version",
+    "[capi][array-schema][version]") {
+  SECTION("- No serialization") {
+    serialize_array_schema_ = false;
+  }
+  SECTION("- Serialization") {
+    serialize_array_schema_ = true;
+  }
+
+  // Create and allocate array schema
+  tiledb_array_schema_t* array_schema;
+  int rc = tiledb_array_schema_alloc(ctx_, TILEDB_SPARSE, &array_schema);
+  REQUIRE(rc == TILEDB_OK);
+
+  // Create dimension
+  tiledb_dimension_t* d;
+  rc = tiledb_dimension_alloc(
+      ctx_, "d", TILEDB_INT64, &DIM_DOMAIN[0], &TILE_EXTENTS[0], &d);
+  REQUIRE(rc == TILEDB_OK);
+
+  // Set domain
+  tiledb_domain_t* domain;
+  rc = tiledb_domain_alloc(ctx_, &domain);
+  REQUIRE(rc == TILEDB_OK);
+  rc = tiledb_domain_add_dimension(ctx_, domain, d);
+  REQUIRE(rc == TILEDB_OK);
+  rc = tiledb_array_schema_set_domain(ctx_, array_schema, domain);
+  REQUIRE(rc == TILEDB_OK);
+
+  // Set attribute
+  tiledb_attribute_t* a;
+  rc = tiledb_attribute_alloc(ctx_, "a", ATTR_TYPE, &a);
+  REQUIRE(rc == TILEDB_OK);
+  rc = tiledb_array_schema_add_attribute(ctx_, array_schema, a);
+  REQUIRE(rc == TILEDB_OK);
+
+  // Instantiate local class
+  SupportedFsLocal local_fs;
+
+  // Create array
+  std::string array_name =
+      local_fs.file_prefix() + local_fs.temp_dir() + "duplicates";
+  create_temp_dir(local_fs.file_prefix() + local_fs.temp_dir());
+  rc = array_create_wrapper(array_name, array_schema);
+  REQUIRE(rc == TILEDB_OK);
+
+  // Clean up
+  tiledb_attribute_free(&a);
+  tiledb_dimension_free(&d);
+  tiledb_domain_free(&domain);
+  tiledb_array_schema_free(&array_schema);
+
+  // Load array schema
+  rc = array_schema_load_wrapper(array_name.c_str(), &array_schema);
+  REQUIRE(rc == TILEDB_OK);
+
+  // Get version.
+  // Version should be constants::format_version because of
+  // ArraySchema::serialize always resets it
```

Comment:
I don't quite follow the "because of ArraySchema::serialize always resets it" comment here, would you mind to clarify? I think the test is correct as written, so maybe this comment can be removed?

Reply:
Fixed.